portal/src/OnboardingRedirect.tsx

```diff
@@ -16,13 +18,15 @@ const OnboardingRedirect: React.VFC = function OnboardingRedirect() {
     if (error != null) {
       return;
     }
-    // redirect to create apps if user doesn't have any apps
-    if (apps && apps.length > 0) {
-      navigate("/");
+    if (
+      (apps === null || apps.length === 0) &&
+      !viewer?.isOnboardingSurveyCompleted
```

Comment:
If viewer is null (`viewer == null`) and is viewer query is still loading, do nothing.